torch/_refs/__init__.py

```diff
@@ -2924,6 +2924,27 @@ def ravel(a: TensorLikeType) -> TensorLikeType:
     return reshape(a, (-1,))
 
 
+# TODO: layout and pin_memory are not supported by refs, but are expected as
+# arguments
+@register_decomposition(torch.ops.aten.new_empty)
+def new_empty(
+    self: TensorLikeType,
+    size: Union[torch.Size, List[int], Tuple[int, ...]],
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
+    layout: Optional[torch.layout] = None,  # unused
+    pin_memory: Optional[bool] = None,  # unused
+    requires_grad: bool = False,
+) -> TensorLikeType:
+    """
+    Reference implementation of torch.Tensor.new_empty
+    """
+    dtype = self.dtype if dtype is None else dtype
+    device = self.device if device is None else device
+
+    return torch.empty(size, dtype=dtype, device=device, requires_grad=requires_grad)
```

Comment:
We should probably test that refs are composite compliant. As written, this ref isn't composite compliant -- e.g. if self is a Tensor subclass, then the returned result isn't a Tensor subclass

Reply:
not a blocker for merging this one, right? eg, `empty_like` also has this behavior:

```py
import torch

class Foo(torch.Tensor):
    def __init__(self):
        super().__init__()

x = torch.tensor([1])
foo = Foo()

x1 = type(torch.empty_like(x))
x2 = type(torch._refs.empty_like(x))

foo1 = type(torch.empty_like(foo))
foo2 = type(torch._refs.empty_like(foo))

print(x1, x2, x1 == x2)
print(foo1, foo2, foo1 == foo2)

# <class 'torch.Tensor'> <class 'torch.Tensor'> True
# <class '__main__.Foo'> <class 'torch.Tensor'> False
```

Reply:
Not a blocker for merging. I'd be curious about other folks opinions; I put up an issue over at https://github.com/pytorch/pytorch/issues/82239

Reply:
note: new_empty ref already landed as part of this: https://github.com/pytorch/pytorch/pull/82049 i didn't notice because it's part of the tensor pack claimed by me on the primtorch tracker. i started working on this a few days ago when there was no prs for it, but then got sidetracked by other refs. the merged commit landed sometime in between